### ISSUE TITLE
Allow UUID type to be returned from a computed column

### ISSUE THREAD

**Issue by pnappa** [hasura/graphql-engine#3507]
Hi team,

When writing a custom function below:

```
CREATE FUNCTION admin_basic_info_id(Info_row "Information")
RETURNS UUID AS $$
    SELECT "UserBasicInfo".id from "UserBasicInfo" 
    INNER JOIN "User" ON "User".user_basic_info_id = "UserBasicInfo".id 
    INNER JOIN "Information" ON "Information".admin_id = "User".id
    WHERE "Information".id = Info_row.id
$$ LANGUAGE sql STABLE;
```

I wanted to add a computed column to expose the id (UUID) of another column that a user cannot access by traversing object relationships due to permissions:

```
query GetAdminInfo($info_id: Int!) {
  Information(where: {id: { _eq: $info_id } } ) {
    id
    admin {  # user querying can't see this object
      info {
        firstName # but would be able to see this one, if it just queried the UserBasicInfo table directly.
      }
    }
  }
}
```

When making a query to add a computed field to Info (via this approximate POST request):

```
{
    "type":"add_computed_field",
    "args":{
        "table":{
            "name":"Information",
            "schema":"public"
        },
        "name":"admin_basic_info_id2",
        "definition":{
            "function":{
				"name":"admin_basic_info_id",
                "schema":"public"
            }
        }
    }
}
```

I got this error:
```
{
  "path": "$.args.definition",
  "error": "the computed field \"admin_basic_info_id2\" cannot be added to table \"Information\" because the function \"admin_basic_info_id\" returning type uuid is not a BASE type",
  "code": "not-supported"
}
```

Judging by a brief look at [this code](https://github.com/hasura/graphql-engine/blob/d8d21c1487c3a67c9a0eec80a0effc74c5cbb2d0/server/src-lib/Hasura/SQL/Types.hs), Hasura doesn't add UUID to it's known Postgres base datatypes.

**Comment by pnappa** [comment]
Also, an aside, when I avoided this by making it work by adding an Int reference, I couldn't seem to add an object relationship based off this? Is adding a relationship based off computed columns supported?

My final workaround was making it a table-computed field, i.e.: https://docs.hasura.io/1.0/graphql/manual/schema/computed-fields.html#table-computed-fields

**Comment by rakeshkky** [comment]
@pnappa We've been very strict about scalar types returned by a computed field. In the next couple of weeks, I'll make the necessary enhancement. 

> Is adding a relationship based off computed columns supported?

Currently No. You can make the best use of table-computed field with the power of input arguments to return related data from another table :slightly_smiling_face: .

**Comment by rakeshkky** [comment]
@pnappa  UUID is added to known postgres types in the [server code](https://github.com/hasura/graphql-engine/blob/04e5a20dc017ddd2e2e5c707aab1d3ddb78526f1/server/src-lib/Hasura/SQL/Types.hs#L351). I assume the issue is fixed from version `v1.1.0`. If still the issue exists please feel free to re-open again. Thank you.
